### ISSUE TITLE
Moving a file to another folder will erase existing file with no confirmation and no backup

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#19990]
Godot 3.0.4

I was reorganizing a project just before starting to version it, but I realized I moved a file in another folder where a file of the same name already existed. Godot erased the old one without asking me, and I cannot find the file in the recycle bin... so I actually lost whatever that file contained.

Please add a dialog for this.

Repro:
1) Have the following file structure:
```
folder/
- camera.gd
- subfolder/
    - camera.gd
```
2) Move `camera.gd` in a folder where the other `camera.gd` is
3) Notice Godot didn't ask anything, nothing in the log, and you lost one of the two files forever

**Comment by vnen** [comment]
Looks the same as #18622 (already solved in master)

**Comment by akien-mga** [comment]
Indeed, duplicate of #18622.
